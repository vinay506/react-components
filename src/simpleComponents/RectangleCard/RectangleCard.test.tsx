import React from 'react';
import { shallow } from 'enzyme';
import { RectangleCard, RoundedBox, TextSecondary ,CommentIcon,NameWithSubText,DisplayWithFlex} from '../index'
import renderer from 'react-test-renderer'


describe('<Rectangle Card>', function () {
    let wrapper: any;
    let props: any;
    beforeEach(() => {
        props = {
            secondaryTextStyles :{
                fontWeight: "500", fontSize: "12px",
                marginBottom: '20px',
                marginTop: '15px',
                width: '300px',
                whiteSpace: 'break-spaces'
            },
            secondaryText:{
                text: `JEE Ready has been helped me so much in my preparation. I practiced all the questions, which were also good and in the right difficulty! I was able to do massively well in JEE`
            },
            avatar:{
                styles:{
                    height: '30px',
                    width: '30px',
                    marginBottom: '20px',
                    marginRight:'10px'
                }
            },
            nameInfo:{
                name: 'Asha Sharma',
                subText: '60 AIR'
            }
        };
    });

    // Snapshot test
    it('render Square Card Wrapper', () => {
        const tree = renderer.create(<RectangleCard></RectangleCard>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should have valid no of childrens in props', () => {
        wrapper = shallow(
            <RectangleCard>
                <CommentIcon styles={{ height: '30px', width: '30px' }}></CommentIcon>
                <TextSecondary text={props.secondaryText.text} styles={props.secondaryTextStyles}></TextSecondary>
                <DisplayWithFlex>
                    <RoundedBox style={props.avatar.styles}>
                    </RoundedBox>
                    <NameWithSubText nameInfo={props.nameInfo}>
                    </NameWithSubText>
                </DisplayWithFlex>
            </RectangleCard>
        );

        const childrenCount = wrapper.children();
        expect(childrenCount).toHaveLength(3)

    })


})