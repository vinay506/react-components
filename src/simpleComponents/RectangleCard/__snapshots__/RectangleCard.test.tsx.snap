// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rectangle Card> render Square Card Wrapper 1`] = `
<div
  className="sc-Axmtr jrBCvi"
/>
`;
